daemon/images/image_list.go

```diff
@@ -59,6 +61,25 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 		return nil, err
 	}
 
+	err = opts.Filters.WalkValues("until", func(value string) error {
+		ts, err := timetypes.GetTimestamp(value, time.Now())
+		if err != nil {
+			return err
+		}
+		seconds, nanoseconds, err := timetypes.ParseTimestamps(ts, 0)
+		if err != nil {
+			return err
+		}
+		timestamp := time.Unix(seconds, nanoseconds)
+		if beforeFilter.IsZero() || beforeFilter.After(timestamp) {
```

Comment:
I was initially looking if we could skip the handling in the `beforeFilter.IsZero()` case, looks like we can't because of the `beforeFilter.After(timestamp)`. We can still optimise in future if we find an alternative.